server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -236,6 +243,7 @@ public Object accumulate(Object accumulated, Object in)
 
       try {
         final Query theQuery = query;
+        final QueryToolChest theToolChest = toolChest;
```

Comment:
Sure, I figured it would probably be outside the scope of this PR, but should be refactored before too many work-arounds are added.